File: problems/problemset/url-router/Rust/template/router.rs
pub struct RouteMatch {
    pub pattern: String,
    pub params: Vec<(String, String)>,
}

pub struct Router {
    // Add your fields here
}

impl Router {
    pub fn new() -> Self {
        todo!()
    }

    pub fn register(&mut self, method: &str, pattern: &str) {
        todo!()
    }

    pub fn match_route(&self, method: &str, url: &str) -> RouteMatch {
        todo!()
    }
}

File: problems/problemset/url-router/C/template/router.c
#include <stdlib.h>
#include <string.h>
#include "Router.h"

struct Router {
    /* add your fields here */
};

Router *router_create(void) {
    Router *r = malloc(sizeof(*r));
    return r;
}

void router_register(Router *r, const char *method, const char *pattern) {
    /* TODO */
    (void)r; (void)method; (void)pattern;
}

RouteMatch router_match(const Router *r, const char *method, const char *url) {
    /* TODO: return a RouteMatch with pattern[0]=='\0' if no match */
    RouteMatch m = {0};
    (void)r; (void)method; (void)url;
    return m;
}

void router_destroy(Router *r) {
    free(r);
}
